extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -105,23 +122,193 @@ export class AmpInstallServiceWorker extends AMP.BaseElement {
     iframe.src = this.iframeSrc_;
     this.element.appendChild(iframe);
   }
+
+  /** @private */
+  maybeInstallUrlRewrite_() {
+    // Only run rewrite in single-doc environment.
+    if (!this.getAmpDoc().isSingleDoc()) {
+      return;
+    }
+
+    const ampdoc = this.getAmpDoc();
+    const win = this.win;
+    const winUrl = parseUrl(win.location.href);
+
+    // Cannot install url rewrite on proxy.
+    if (isProxyOrigin(winUrl)) {
```

Comment:
Does this means it can't rewrite when served from the cache? Or does proxy stand for something different here?